app/models/reply.rb

```diff
@@ -118,4 +118,15 @@ def update_post_authors
     return if is_import
     post.user_joined(user)
   end
+
+  def remove_post_author
+    return if post.user == user
+    return unless post.replies.where(user: user).blank?
+    post_author = post.post_authors.find_by(user: user)
+    if post.authors_locked == false
```

Comment:
Can we do this based off `can_owe` again?

```ruby
if post_author.can_owe?
  post_author.update_attributes(joined: false)
else
  post_author.destroy
end
```

Reply:
joining a thread stets can_owe to true

Reply:
in that case, can we still flip this if-conditional:

```ruby
if post.authors_locked?
  post_author.update_attributes(joined: false)
else
  post_author.destroy
end
```